src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -44,10 +76,11 @@ public async Task<IEnumerable<EmployerInfo>> GetLegalEntitiesForProviderAsync(lo
                     if (!response.IsSuccessStatusCode)
                     {
                         _logger.LogError("An invalid response received when trying to get provider relationships");
-                        return new EmployerInfo[]{};
-                    }                
-                    var content  = await response.Content.ReadAsStringAsync();
-                    providerPermissions = JsonConvert.DeserializeObject<ProviderPermissions>(content);
+                        return null;
```

Comment:
I am not fan of returning nulls as far as possible, especially when the return value is a list. I saw where this is being consumed and we can easily check for empty list in both the cases.

Reply:
👍   I'll return a `new ProviderPermissions { AccountProviderLegalEntities = Enumerable.Empty<LegalEntityDto>() };` then we won't need to do a null check